src/controller/app_routes.rs

```diff
@@ -329,4 +335,32 @@ impl AppRoutes {
         tracing::info!("[Middleware] Adding timeout layer");
         app
     }
+
+    fn add_server_header(
+        app: AXRouter<AppContext>,
+        config: &config::Server,
+    ) -> AXRouter<AppContext> {
+        let ident_value =
+            config
+                .ident
+                .as_ref()
+                .map_or(DEFAULT_IDENT_HEADER_VALUE.clone(), |ident| {
+                    match http::header::HeaderValue::from_str(ident) {
+                        Ok(val) => val,
+                        Err(e) => {
+                            tracing::info!(
+                                error = format!("{}", e),
+                                val = ident,
+                                "could not set custom ident header"
+                            );
+                            DEFAULT_IDENT_HEADER_VALUE.clone()
+                        }
+                    }
+                });
+
+        app.layer(SetResponseHeaderLayer::overriding(
+            http::header::SERVER,
```

Comment:
lets do `X-Powered-By` instead of `Server`, from what I see `Server` has a chance to get overridden by edge or something like ngnix, etc.